File: weather-mvp-dagger/src/main/java/com/baronzhang/android/weather/feature/home/HomePageModule.java
package com.baronzhang.android.weather.feature.home;

import com.baronzhang.android.weather.di.ActivityScoped;
import com.baronzhang.android.weather.feature.contract.HomePageContract;

import dagger.Module;
import dagger.Provides;

/**
 * This is a Dagger module. We use this to pass in the View dependency to the
 * {@link HomePagePresenter}
 *
 * @author baronzhang (baron[dot]zhanglei[at]gmail[dot]com)
 *         2016/11/30
 */
@Module
public class HomePageModule {

    private HomePageContract.View view;

    public HomePageModule(HomePageContract.View view) {
        this.view = view;
    }

    @Provides
    @ActivityScoped
    HomePageContract.View provideHomePageContractView() {
        return view;
    }

}

File: weather-mvp-dagger-kotlin/src/main/java/com/minimalist/weather/kotlin/view/home/HomePageModule.kt
package com.minimalist.weather.kotlin.view.home


import com.minimalist.weather.kotlin.di.ActivityScoped
import com.minimalist.weather.kotlin.di.FragmentScoped
import com.minimalist.weather.kotlin.view.contract.HomePageContract
import dagger.Binds
import dagger.Module
import dagger.android.ContributesAndroidInjector

/**
 * This is a Dagger module. We use this to pass in the View dependency to the
 * [HomePagePresenter]
 */
@Module
abstract class HomePageModule {

    @FragmentScoped
    @ContributesAndroidInjector
    abstract fun homePageFragment(): HomePageFragment

    @ActivityScoped
    @Binds
    abstract fun homePagePresenter(presenter: HomePagePresenter): HomePageContract.Presenter

}
